src/client/components/atoms/TextInput/TextInput.jsx

```diff
@@ -39,13 +41,14 @@ const TextInput = (props) => {
   return (
     <div className={container}>
       <input
+        name={inputName}
         className={input}
         id={id}
         value={value}
         onChange={onChange}
         required={required}
         disabled={disabled}
-        type='text'
+        type={type || 'text'}
```

Comment:
I can figure out the reason you changed this was because of the needing of passing it a password type value or maybe, email? This input is only suppose to support text type formats, which includes both all of the named above. But, this wont solve it as it open the possibility to any type. My suggestions are this:

1. Change ```type={type || 'text'}``` for ```type={type}```
2. Add to types the following PropType: ```PropType.oneOf([])```, the array represent the strings we're supporting here, for example: ```[...'text', 'password']```
3. Do not make this prop required and add the following to defaultProps: ```type: 'text'```

Reply:
Thank you for your suggestion